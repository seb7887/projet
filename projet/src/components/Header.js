import React from 'react';

import Logo from '../images/rocket.svg';

const headerStyles = {
  display: 'flex',
  flexDirection: 'row',
  padding: '1rem',
};

const h1Styles = {
  textTransform: 'uppercase',
  fontSize: '2rem',
  letterSpacing: '.5rem',
  marginLeft: '2rem',
  marginTop: '1rem',
};

const logoStyles = {
  height: '4rem',
  display: 'inline-block',
  filter: 'opacity(.5) drop-shadow(0 0 0 #ffffff)',
};

class Header extends React.Component {
  shouldComponentUpdate(nextProps, nextState) {
    return false;
  }

  render() {
    return (
      <div style={headerStyles}>
        <img src={Logo} alt='Projet Home' style={logoStyles} />
        <h1 style={h1Styles}>Projet</h1>
      </div>
    );
  }
}

export default Header;
