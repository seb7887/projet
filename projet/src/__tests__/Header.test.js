import React from 'react';
import { shallow } from 'enzyme';

import Header from '../components/Header';

describe('Header Component', () => {
  it('expect to render Header component', () => {
    expect(shallow(<Header />)).toMatchSnapshot();
  })

  it('expect not to be updated', () => {
    const component = shallow(<Header />);
    const shouldUpdate = component.instance().shouldComponentUpdate();
    expect(shouldUpdate).toBe(false);
  })
})
