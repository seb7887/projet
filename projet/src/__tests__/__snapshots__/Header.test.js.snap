// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component expect to render Header component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Projet Home"
          src="rocket.svg"
          style={
                    Object {
                              "display": "inline-block",
                              "filter": "opacity(.5) drop-shadow(0 0 0 #ffffff)",
                              "height": "4rem",
                            }
          }
/>,
        <h1
          style={
                    Object {
                              "fontSize": "2rem",
                              "letterSpacing": ".5rem",
                              "marginLeft": "2rem",
                              "marginTop": "1rem",
                              "textTransform": "uppercase",
                            }
          }
>
          Projet
</h1>,
      ],
      "style": Object {
        "display": "flex",
        "flexDirection": "row",
        "padding": "1rem",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Projet Home",
          "src": "rocket.svg",
          "style": Object {
            "display": "inline-block",
            "filter": "opacity(.5) drop-shadow(0 0 0 #ffffff)",
            "height": "4rem",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Projet",
          "style": Object {
            "fontSize": "2rem",
            "letterSpacing": ".5rem",
            "marginLeft": "2rem",
            "marginTop": "1rem",
            "textTransform": "uppercase",
          },
        },
        "ref": null,
        "rendered": "Projet",
        "type": "h1",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Projet Home"
            src="rocket.svg"
            style={
                        Object {
                                    "display": "inline-block",
                                    "filter": "opacity(.5) drop-shadow(0 0 0 #ffffff)",
                                    "height": "4rem",
                                  }
            }
/>,
          <h1
            style={
                        Object {
                                    "fontSize": "2rem",
                                    "letterSpacing": ".5rem",
                                    "marginLeft": "2rem",
                                    "marginTop": "1rem",
                                    "textTransform": "uppercase",
                                  }
            }
>
            Projet
</h1>,
        ],
        "style": Object {
          "display": "flex",
          "flexDirection": "row",
          "padding": "1rem",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Projet Home",
            "src": "rocket.svg",
            "style": Object {
              "display": "inline-block",
              "filter": "opacity(.5) drop-shadow(0 0 0 #ffffff)",
              "height": "4rem",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Projet",
            "style": Object {
              "fontSize": "2rem",
              "letterSpacing": ".5rem",
              "marginLeft": "2rem",
              "marginTop": "1rem",
              "textTransform": "uppercase",
            },
          },
          "ref": null,
          "rendered": "Projet",
          "type": "h1",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
